Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h

```diff
@@ -1344,6 +1392,7 @@ refine_balls()
   Triangulation& tr = c3t3_.triangulation();
 
   // Loop
+  bool check_edge_distance = !std::is_same_v<Distance_Function, NoDistanceFunction<FT>>;
   bool restart = true;
```

Comment:
I do not think that boolean check works. `Distance_function` is never equal to `NoDistanceFunction<FT>`, if I have correctly understood the code.

Reply:
If `Protect_edges_sizing_field` is created without a `DistanceFunction`, then, `Distance_function` will equal `NoDistanceFunction<FT>`.

One example of this is present in the test `test_meshing_polylines_only.cpp` :
```cpp
// Mesh criteria
Mesh_criteria criteria(edge_size = 0.1);
typedef Mesh_criteria::Edge_criteria Edge_criteria;
typedef CGAL::Mesh_3::internal::Edge_criteria_sizing_field_wrapper<Edge_criteria> Sizing_field;
CGAL::Mesh_3::Protect_edges_sizing_field<C3t3, Mesh_domain, Sizing_field>
  protect_edges(c3t3, domain, Sizing_field(criteria.edge_criteria_object()), 0.01);
protect_edges(true);
```
In this example, no `DistanceFunction` is set, so we do not want to check the edge distance.

Reply:
Coming back on this : in most cases, `Protect_edges_sizing_field` is automatically created when calling `make_mesh_3` with feature protection activated. In this case, the `Distance_field` passed to `Protect_edges_sizing_field` is a wrapper and is not `NoDistanceFunction<FT>`, so the check is indeed useless in most cases.
In an ideal scenario, we want to make the boolean `check_edge_distance` equal `false` when using the default `Distance_field` parameter and calling `make_mesh_3`. But for now, `make_mesh_3` cannot know whether or not the `MeshEdgeCriteria_3` have a default `Distance_field`.